nrf_rpc/include/nrf_rpc_cbor.h

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+
+#ifndef _NRF_RPC_CBOR_H_
+#define _NRF_RPC_CBOR_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#include <tinycbor/cbor.h>
+#include <tinycbor/cbor_buf_writer.h>
+#include <tinycbor/cbor_buf_reader.h>
+
+#include <nrf_rpc.h>
+
+
+/**
+ * @defgroup nrf_rpc_cbor TinyCBOR serialization layer for nRF RPC.
+ * @{
+ * @ingroup nrf_rpc
+ *
+ * @brief Module simplifying usage of TinyCBOR as a serialization for nRF RPC
+ * module.
+ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** @brief Callback that handles decoding of commands, events and responses.
+ *
+ * @param value        TinyCBOR value to decode.
+ * @param handler_data Custom handler data.
+ */
+typedef void (*nrf_rpc_cbor_handler_t)(CborValue *value, void *handler_data);
+
+
+/* Structure used internally to define TinCBOR command or event decoder. */
+struct _nrf_rpc_cbor_decoder {
+	nrf_rpc_cbor_handler_t handler;
+	void *handler_data;
+	bool decoding_done_required;
+};
+
+
+/** @brief Context for encoding and sending commands, events and responses.
+ *
+ * Initialize it with @ref NRF_RPC_CBOR_ALLOC macro. Only `encoder` field is
+ * significant for the API, other fields are internal.
+ */
+struct nrf_rpc_cbor_ctx {
+	/** @brief TinyCBOR encoder. */
+	CborEncoder encoder;
+	struct cbor_buf_writer writer;
+	uint8_t *out_packet;
+};
+
+
+/** @brief Context for encoding commands, sending commands, receiving responses
+ * and parsing them.
+ *
+ * Initialize it with @ref NRF_RPC_CBOR_ALLOC macro. Only `encoder` and `value`
+ * fields are significant for the API, other fields are internal.
+ */
+struct nrf_rpc_cbor_rsp_ctx {
+	union {
+		struct {
+			/** @brief TinyCBOR encoder for encoding command. */
+			CborEncoder encoder;
+			struct cbor_buf_writer writer;
+			uint8_t *out_packet;
+		};
+		struct {
+			/** @brief TinyCBOR value for parsing response. */
+			CborValue value;
+			CborParser parser;
+			struct cbor_buf_reader reader;
+			const uint8_t *in_packet;
+		};
+	};
+};
+
+
+/** @brief Register a command decoder.
+ *
+ * @param _group   Group that the decoder will belong to, created with a
+ *                 @ref NRF_RPC_GROUP_DEFINE().
+ * @param _name    Name of the decoder.
+ * @param _cmd     Command id. Can be from 0 to 254.
```

Comment:
255 limitation is per single group. API containing more than 255 functions will be really huge, so it will be probably reasonable to divide it to smaller groups.

Another, more important reason to forcing low number of IDs per group is performance. ID at some point must be converted to actual function. Searching matching function among huge number of registered functions will take some time.

Reply:
I understand, I see that others also asked this question :) Thank you.